include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,518 @@
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_increase_bin_number_to.cpp
+     */
+    void increase_bin_number_to(bin_count const new_bins)
+    //!\cond
+        requires data_layout_mode == data_layout::uncompressed
+    //!\endcond
+    {
+        if (new_bins.get() < bins_)
+            throw std::invalid_argument{"The value of new_bins must be >= the current number of bins."};
+
+        // Equivalent to ceil(new_bins.get() / 64)
+        size_t new_bin_width = (new_bins.get() + 63) >> 6;
+
+        bins_ = new_bins.get();
+        result_buffer.resize(bins_);
+
+        if (new_bin_width == bin_width) // No need for internal resize if bin_width does not change.
+            return;
+
+        size_t new_block_size = new_bin_width << 6;
+        size_t new_bits       = block_count * new_block_size;
+
+        size_t idx_{new_bits}, idx{data.size()};
+        size_t delta = new_block_size - block_size + 64;
+
+        data.resize(new_bits);
+
+        for (size_t i = idx_, j = idx; j > 0; i -= new_block_size, j -= block_size)
+        {
+            size_t stop = i - new_block_size;
+
+            for (size_t ii = i - delta, jj = j - 64; stop && ii >= stop; ii -= 64, jj -= 64)
+            {
+                uint64_t old = data.get_int(jj);
+                data.set_int(jj, 0);
+                data.set_int(ii, old);
+            }
+        }
+
+        bin_width = new_bin_width;
+        block_size = new_block_size;
+    }
+    //!\}
+
+    /*!\name Lookup
+     * \{
+     */
+    /*!\brief Determines set membership of a given value.
+     * \param[in] value The raw value to process.
+     *
+     * \attention The result of this function must always be bound via reference, e.g. `auto &`.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_bulk_contains.cpp
+     */
+    [[nodiscard]] binning_bitvector const & bulk_contains(size_t const value) const & noexcept
+    {
+        assert(result_buffer.size() == bins());
+
+        std::vector<size_t> idx = pre_hash;
+
+        std::ranges::for_each(idx, [this, value] (size_t & i) { i *= value; hash_and_fit(i); });
+
+        for (size_t batch = 0; batch < bin_width; ++batch)
+        {
+           assert(idx[0] < data.size());
+           size_t tmp = data.get_int(idx[0]);
+           idx[0] += 64;
+
+           for (size_t i = 1; i < hash_funs; ++i)
+           {
+               assert(idx[i] < data.size());
+               tmp &= data.get_int(idx[i]);
+               idx[i] += 64;
+           }
+
+           result_buffer.set_int(batch<<6, tmp);
+        }
+
+        return result_buffer;
+    }
+
+    // `bulk_contains` cannot be called on a temporary, since the the object the returned reference points to
+    // is immediately destroyed.
+    [[nodiscard]] binning_bitvector const & bulk_contains(size_t const value) const && noexcept = delete;
```

Comment:
Not really needed, but can prevent misuse, e.g. `seqan3::interleaved_bloom_filter{/*ctor args*/}.bulk_contains(23);`